### ISSUE TITLE
[SHFB: Error BE0043] NullReferenceException during the build

### ISSUE THREAD

**Issue by koszeggy** [EWSoftware/SHFB#940]
Though I could successfully build the help with the latest release for my largest project but it did not work for a [smaller one](https://github.com/koszeggy/KGySoft.Json).

**Log:**
```xml
<buildStep step="GenerateReflectionInfo">
Generating reflection information...
[C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\amd64\MSBuild.exe - GenerateRefInfo-0d31f5c5-1e48-438d-ac56-6a221330d99d.proj]
Build started 2022. 09. 21. 17:22:32.
Project &quot;D:\Nyelvek\CSharp\KGySoft\KGySoft.Json\KGySoft.Json\HelpOut\Working\GenerateRefInfo-0d31f5c5-1e48-438d-ac56-6a221330d99d.proj&quot; on node 1 (default targets).
PrepareForBuild:
  Creating directory &quot;obj\Debug\&quot;.
GenerateBindingRedirects:
  No suggested binding redirects from ResolveAssemblyReferences.
GenerateRefInfo:
  Microsoft.Build (v17.2.1+52cd2da314670cc504a877e251a0728124e7cf07)
  c Microsoft Corporation. All rights reserved.
  No source code context base path specified.  Source context information is unavailable.
  Loaded 1 assemblies for reflection and 168 dependency assemblies.
MRefBuilder : error : An unexpected error occurred trying to execute the MRefBuilder MSBuild task: System.NullReferenceException: Object reference not set to an instance of an object. [D:\Nyelvek\CSharp\KGySoft\KGySoft.Json\KGySoft.Json\HelpOut\Working\GenerateRefInfo-0d31f5c5-1e48-438d-ac56-6a221330d99d.proj]
MRefBuilder : error :    at Sandcastle.Tools.ExtensionMethodAddIn.RecordExtensionMethods(XmlWriter writer, Object info) [D:\Nyelvek\CSharp\KGySoft\KGySoft.Json\KGySoft.Json\HelpOut\Working\GenerateRefInfo-0d31f5c5-1e48-438d-ac56-6a221330d99d.proj]
MRefBuilder : error :    at Sandcastle.Tools.ManagedReflectionWriter.StartElementCallbacks(String name, Object info) [D:\Nyelvek\CSharp\KGySoft\KGySoft.Json\KGySoft.Json\HelpOut\Working\GenerateRefInfo-0d31f5c5-1e48-438d-ac56-6a221330d99d.proj]
MRefBuilder : error :    at Sandcastle.Tools.ManagedReflectionWriter.VisitNamespaces(NamespaceList spaces) [D:\Nyelvek\CSharp\KGySoft\KGySoft.Json\KGySoft.Json\HelpOut\Working\GenerateRefInfo-0d31f5c5-1e48-438d-ac56-6a221330d99d.proj]
MRefBuilder : error :    at Sandcastle.Tools.Reflection.ApiVisitor.VisitApis() [D:\Nyelvek\CSharp\KGySoft\KGySoft.Json\KGySoft.Json\HelpOut\Working\GenerateRefInfo-0d31f5c5-1e48-438d-ac56-6a221330d99d.proj]
MRefBuilder : error :    at Sandcastle.Tools.MSBuild.MRefBuilder.GenerateReflectionInformation() [D:\Nyelvek\CSharp\KGySoft\KGySoft.Json\KGySoft.Json\HelpOut\Working\GenerateRefInfo-0d31f5c5-1e48-438d-ac56-6a221330d99d.proj]
MRefBuilder : error :    at Sandcastle.Tools.MSBuild.MRefBuilder.Execute() [D:\Nyelvek\CSharp\KGySoft\KGySoft.Json\KGySoft.Json\HelpOut\Working\GenerateRefInfo-0d31f5c5-1e48-438d-ac56-6a221330d99d.proj]
Done Building Project &quot;D:\Nyelvek\CSharp\KGySoft\KGySoft.Json\KGySoft.Json\HelpOut\Working\GenerateRefInfo-0d31f5c5-1e48-438d-ac56-6a221330d99d.proj&quot; (default targets) -- FAILED.
    Last step completed in 00:00:00,9650
</buildStep>
<buildStep step="Failed">

SHFB: Error BE0043: Unexpected error detected in last build step.  See build log for details.
   at SandcastleBuilder.Utils.BuildEngine.TaskRunner.Run(String processFilename, String targetFile, String arguments)
   at SandcastleBuilder.Utils.BuildEngine.BuildProcess.Build()

</buildStep>
```

Not sure whether they are related but maybe can be caused by the fix of #925 as the exception comes from `RecordExtensionMethods` and this is the project where the [extension methods](https://docs.kgysoft.net/json/?topic=html/T_KGySoft_Json_JsonValueExtensions.htm) did not appear for a [value type](https://docs.kgysoft.net/json/?topic=html/T_KGySoft_Json_JsonValue.htm).

I apologize for not being able to provide a minimal repro, the example of #925 is now working perfectly with the latest release so there must be something special in my case. Feel free to clone [this](https://github.com/koszeggy/KGySoft.Json) repo if the issue is not obvious from the posted stack trace. I created a [new help project file](https://github.com/koszeggy/KGySoft.Json/blob/HelpVS2022Style/KGySoft.Json/KGySoft.Json.VS2022.shfbproj) on a new [HelpVS2022Style](https://github.com/koszeggy/KGySoft.Json/tree/HelpVS2022Style) branch in case you need to investigate it.

**Comment by EWSoftware** [comment]
I believe this is related to #931 which was also caused by the fix in #925.  It's fixed for the next release and your branch built as expected with it.  I've got some other pending changes which I'll wrap up and publish soon.

**Comment by koszeggy** [comment]
Ah ok, sounds promising, thank you!
